### PR TITLE
Initial commit for creator lock component and stories

### PR DESCRIPTION
**Pull request by benwerd**
This PR contains the saved and confirming states for the lock from the creator side.

There are a few known issues that will need to be added in a following PR:

* The edit and code icons are broken
* All icons are missing their grey background
* The icon bar on saved locks does not show/hide

I've also added TODOs for the exchange rate values of the lock, and have kept the styles for those values commented out for now.

### CODE REVIEW

**Comment by benwerd** [comment]
screen shot 2018-09-13 at 1 38 52 pm
screen shot 2018-09-13 at 1 38 00 pm

File: unlock-app/src/components/creator/CreatorLock.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+  // Some sanitization of strings to display
+  let name = lock.name || 'New Lock'
+  let displayAddress = lock.address ? lock.address.substr(0, 16) : ''
+  let outstandingKeys = lock.maxNumberOfKeys - lock.outstandingKeys || 0
+
+  if (status === 'deployed') { // the transaction was mined and confirmed at least 12 times
+    // TODO add USD values to lock
+    // TODO add all-time balance to lock
+    return (
+      <CreatorLockRow>
```
Comment (julien51) [inline unlock-app/src/components/creator/CreatorLock.js]: Would it be worth splitting that in multiple files eventually? All importing the same elements from below?

File: unlock-app/src/components/creator/CreatorLock.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+import Duration from '../helpers/Duration'
+import Balance from '../helpers/Balance'
+import Icon from '../lock/Icon'
+
+import Icons from '../interface/icons'
+
+export function CreatorLock({ lock, status = 'deployed' }) {
+  // Some sanitization of strings to display
+  let name = lock.name || 'New Lock'
+  let displayAddress = lock.address ? lock.address.substr(0, 16) : ''
```
Comment (julien51) [inline unlock-app/src/components/creator/CreatorLock.js]: Wouldn;t it be better to use CSS for this?

File: unlock-app/src/components/creator/CreatorLock.js (hunk 1 above)
Comment (julien51) [reply unlock-app/src/components/creator/CreatorLock.js]: Also, not sure it is worth prefixing `Creator` on all of these components?

File: unlock-app/src/components/helpers/Balance.js [hunk 3]
```diff
@@ -7,15 +7,19 @@ import Web3Utils from 'web3-utils'
  * @param {*} amount: the amount to convert to Eth
  * @param {string} unit: the unit of the amount to convert to Eth
  */
-export function Balance({ amount, unit = 'wei' }) {
-  const inWei = Web3Utils.toWei(amount || '0', unit)
-  const inEth = Web3Utils.fromWei(inWei, 'ether')
-  return (<span>Ξ {inEth}</span>)
+export function Balance({ amount, unit = 'wei', symbol=true }) {
```
Comment (julien51) [inline unlock-app/src/components/helpers/Balance.js]: Do you mind adding a test for this? (I think the Balance component has tests?)

File: unlock-app/src/components/interface/icons/Eth.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import React from 'react'
 
 const Eth = props => (
-  <svg viewBox="0 0 7 11" width="1em" height="1em" {...props}>
```
Comment (julien51) [inline unlock-app/src/components/interface/icons/Eth.js]: nope... these are automatically generated so we cannot change the values in there (I need to find a way to indicate that).
If you need the icons to be larger just change the `font-size` of the element in which the icons are displayed.

Comment (benwerd) [reply unlock-app/src/components/interface/icons/Eth.js]: Ah got it. OK!

File: unlock-app/src/components/helpers/Balance.js (hunk 3 above)
Comment (benwerd) [reply unlock-app/src/components/helpers/Balance.js]: 👍

File: unlock-app/src/components/creator/CreatorLock.js (hunk 1 above)
Comment (benwerd) [reply unlock-app/src/components/creator/CreatorLock.js]: I wanted to keep the namespace clear for user-facing components, as these are all creator-facing UI elements. But I could change some of the names?

**Comment by benwerd** [comment]
Updated. New screenshots: 

screen shot 2018-09-13 at 1 54 41 pm
screen shot 2018-09-13 at 1 54 36 pm

File: unlock-app/src/components/creator/CreatorLock.js (hunk 1 above)
Comment (benwerd) [reply unlock-app/src/components/creator/CreatorLock.js]: Could totally split this up into multiple files though.

Comment (benwerd) [reply unlock-app/src/components/creator/CreatorLock.js]: I'll do exactly that in a follow-up PR.

**Comment by benwerd** [comment]
Okay. Tests and reformatting completed. I'll follow up with splitting up into smaller components. PTAL.

File: unlock-app/src/components/creator/CreatorLock.js (hunk 1 above)
Comment (julien51) [reply unlock-app/src/components/creator/CreatorLock.js]: About the naming, I think one of the great thing about the styled components approach is that everything is scoped basically. Thesr components should but be reused.

**Pull request merged**